### DISCUSSION TITLE
Can Watchtower perform a build and restart rather than just a restart?

### DISCUSSION THREAD

**Discussion by kevdogg** [containrrr/watchtower#692] [General] [answered]
If I have a base container -- for example nginx:latest and create a custom Dockerfile to include some packages and install scripts, can watchtower actually update from the base container and then run a docker build using the Dockerfile? 

I suppose a workaround would be to create a repository that uses a base image and the repository automatically gets updated when the upstream is updated and then watchtower pulls from this.  

I'm just wondering what the best practices are in this case since it seems like adding a few things to a base image and using Dockerfiles are pretty common.

**Comment by simonaronsson** [comment, marked as answer]
No, sorry. 

> I suppose a workaround would be to create a repository that uses a base image and the repository automatically gets updated when the upstream is updated and then watchtower pulls from this.

This would indeed be the solution I'd recommend for this. 👍

**Comment by DevDavido** [comment]
Exactly the same problem I was having too @kevdogg! What I eventually did is to search the Dockerfiles for the `FROM` declarations and then check for updates using [this script](https://gitlab.com/MatthiasLohr/omnibus-gitlab-management-scripts/-/blob/main/docker-image-update-check.sh). Of course, there are other ways to use base images in Dockerfiles, for example with `COPY --from=IMAGE ...`, so there is still room for improvement with this solution. In case of image updates/non-existence, the image is pulled and if at least 1 update exists, the corresponding service(s) are rebuilt using docker compose. Watchtower is then used in no-pull mode to restart the containers (`watchtower --no-pull ...`). You can find my script here: https://gist.github.com/DevDavido/1a0e7d3a49cff97ae7c044732d2697d9

**Reply by koinupapi** [reply]
@DevDavido 

Hi David!
So, before I went further I wanted to ask about your scripts and if it would be beneficial to me for my use-case.

I'm relatively new to dockerfile and the like.

The problem I'm having is i manually downloaded the latest mkdocs-material image, and then used dockerfile to customize it.

I spun up a container for the website using Portainer Stacks (they're name for Docker Compose). All is working.

I want to use Watchtowerr to tell me when there is an update to the base image, mkdocs-material.

But right now, every hour it thinks there is a new update to the image.

I don't know why, but I'm fairly certain I did something wrong here.

If I didn't do something wrong, then I guess it isn't part of Watchtower functionality, in which case, does the script you mentioned help me?

Thank you for reading! :)
And also, if there is any more info you need from me, please let me know!

**Reply by DevDavido** [reply]
Hi @koinupapi,

Based on the provided information it seems odd that Watchtowerr thinks there's an update to the image every hour. Maybe check further if the configuration or setup is correct in that case.

The script I provided together with the [`docker-image-update-check.sh`](https://gitlab.com/MatthiasLohr/omnibus-gitlab-management-scripts/-/blob/main/docker-image-update-check.sh) is focusing on updates of custom Dockerfile base images, which checks for updates of those base images. If there are updates, it would pull those updated base images and rebuild them in the context of docker compose. At first sight, it seems that this would be a more suitable solution for your setup.

Hope I could help a bit with your question.

**Reply by koinupapi** [reply]
Thanks for the reply! @DevDavido 

I'm gonna start with your suggestion and probably reset up my Watchtower instance, and go from there.

I asked in Reddit as well, and they suggested that if i just turn off the auto-update for this container, and get RSS feed notifications lol

That's another option, so i think I'll just have to weigh my options and figured out what i want at the end of the day.

Thanks so much for your help!
